homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,176 @@
+STATE_PRESET_FREEZE = "freezeMode"
+STATE_PRESET_MANUAL = "manualMode"
+STATE_PRESET_SLEEPING_MODE = "sleepingMode"
+STATE_PRESET_SUDDEN_DROP_MODE = "suddenDropMode"
+
+OVERKIZ_TO_HVAC_MODES: dict[str, str] = {
+    STATE_DEROGATION_ACTIVE: HVAC_MODE_HEAT,
+    STATE_DEROGATION_INACTIVE: HVAC_MODE_AUTO,
+}
+OVERKIZ_TO_PRESET_MODES: dict[str, str] = {
+    STATE_PRESET_AT_HOME: PRESET_HOME,
+    STATE_PRESET_AWAY: PRESET_AWAY,
+    STATE_PRESET_FREEZE: PRESET_FREEZE,
+    STATE_PRESET_MANUAL: PRESET_NONE,
+    STATE_PRESET_SLEEPING_MODE: PRESET_NIGHT,
+    STATE_PRESET_SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: "somfythermostat:AtHomeTargetTemperatureState",
+    PRESET_AWAY: "somfythermostat:AwayModeTargetTemperatureState",
+    PRESET_FREEZE: "somfythermostat:FreezeModeTargetTemperatureState",
+    PRESET_NIGHT: "somfythermostat:SleepingModeTargetTemperatureState",
+}
+
+
+class SomfyThermostat(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+    _attr_hvac_modes = [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_min_temp = 15.0
+    _attr_max_temp = 26.0
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODES[
+            cast(
+                str, self.executor.select_state(OverkizState.CORE_DEROGATION_ACTIVATION)
+            )
+        ]
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        if not self.current_temperature or not self.target_temperature:
+            return CURRENT_HVAC_IDLE
+        if self.current_temperature < self.target_temperature:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return OVERKIZ_TO_PRESET_MODES[
+                cast(
+                    str,
+                    self.executor.select_state(
+                        OverkizState.SOMFY_THERMOSTAT_HEATING_MODE
+                    ),
+                )
+            ]
+        return OVERKIZ_TO_PRESET_MODES[
+            cast(
+                str,
+                self.executor.select_state(
+                    OverkizState.SOMFY_THERMOSTAT_DEROGATION_HEATING_MODE
+                ),
+            )
+        ]
```

Comment:
```suggestion
        if self.hvac_mode == HVAC_MODE_AUTO:
            state_key = OverkizState.SOMFY_THERMOSTAT_HEATING_MODE
        else:
            state_key = OverkizState.SOMFY_THERMOSTAT_DEROGATION_HEATING_MODE
        
        return OVERKIZ_TO_PRESET_MODES[
            cast(
                str,
                self.executor.select_state(state_key),
            )
        ]
```

Perhaps we can simplify it a bit? The other way (due to the cast) doesn't make it very easy to read.